### PR TITLE
Python/WinRT: await IAsync* support

### PR DESCRIPTION
**Pull request by devhawk**
This PR adds support for directly awaiting an IAsync* instance 

generated code available: https://github.com/devhawk/pywinrt-output/tree/await-iasync

### CODE REVIEW

File: src/test/python/complete_winml_tutorial.py [hunk 1]
```diff
@@ -68,10 +41,10 @@ def load_model(model_path):
     from pyrt.windows.graphics.imaging import BitmapDecoder
     from pyrt.windows.media import VideoFrame
 
-    file = await wrap_async_op(StorageFile.GetFileFromPathAsync(os.fspath(file_path)))
-    stream = await wrap_async_op(file.OpenAsync(0)) # 0 == FileAccessMode::Read 
-    decoder = await wrap_async_op(BitmapDecoder.CreateAsync(stream))
-    software_bitmap = await wrap_async_op(decoder.GetSoftwareBitmapAsync())
+    file = await StorageFile.GetFileFromPathAsync(os.fspath(file_path))
```
Comment (kennykerr) [inline src/test/python/complete_winml_tutorial.py]: Pretty. 😍

Comment (devhawk) [reply src/test/python/complete_winml_tutorial.py]: #naturalAndFamiliar

File: src/tool/python/strings/pybase.h [hunk 2]
```diff
@@ -808,4 +808,73 @@ namespace py
...
+        else
+        {
+            return convert(operation.GetResults());
+        }
+    }
+
+    template <typename Async>
+    PyObject* dunder_await(Async const& async)
+    {
+        PyObject* loop = PyObject_CallMethodObjArgs(PyImport_ImportModule("asyncio"), PyUnicode_FromString("get_event_loop"), nullptr);
```
[the remark this answers is not in the document]
Comment (zooba) [reply src/tool/python/strings/pybase.h]: That said, `PyObject_CallMethod(asyncio_module, "get_event_loop", nullptr)` is the same, and will handle at least the string conversion. (Pass a format string of one `"O"` for each `PyObject*` you're passing in, or a null format string for no args)

File: src/test/python/testgeoloc.py [hunk 3]
```diff
@@ -2,6 +2,17 @@
...
 
+# async_test inspired by https://stackoverflow.com/a/23036785
+def async_test(test):
+    def wrapper(*args, **kwargs):
+        original_loop = asyncio.get_event_loop()
+        test_loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(test_loop)
+        test_loop.run_until_complete(test(*args, **kwargs))
+        test_loop.close()
+        asyncio.set_event_loop(original_loop)
```
[the remark this answers is not in the document]
Comment (devhawk) [reply src/test/python/testgeoloc.py]: will do

**Pull request merged**